silk/apps.py

```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+class SilkAppConfig(AppConfig):
+    default_auto_field = 'django.db.models.AutoField'
```

Comment:
I just checked readme, as you suggest it tells user to run makemigration. This looks like an anti-pattern to me. Not sure why it is required at first place? If required then we should correct this.

Migration should be the responsibility of the package as it defines and maintains its own models. There are number of problems that could occurs if not done that way - for example any migration generated for silk won't be committed to version control.

I have added additional test case in this PR to make sure this doesn't happen.

Reply:
I agree that we should investigate and make any changes required to remove makemigrations from our setup.

I won't be available to check up on this for another week. But, if I recall correctly the only thing I saw makemigrations do was create the auto field, which is fairly pointless and can/should be manually committed.

Unfortunately the repo leadership isn't available to tell us why they went down this route.

Reply:
I checked blame - it was introduced with #113

Reply:
it was right, you are not getting it right, initially 3rd  party migrations framework was used, that PR just switched to django new schema migration framework.

Reply:
@auvipy Do you recall any particular reason why the README suggests running `makemigrations` ?

To me it is not necessary at all. Moreover added test in this PR that will raise red flag when there are pending migration inside the package.

Reply:
bigAutoField will be used for new django projects autogen primary key, that should not effect this package, for the change we had we should generate new migrations which will need to be supplied with this package.

Reply:
also you can follow he docs https://docs.djangoproject.com/en/3.2/ref/applications/

Reply:
@Archmonger What additional components in this PR (except for README change, for which another PR could have been filed) would have been needed in it?

As per the review done, if we do those changes then this PR would be identical to #495.

I only wrote this PR because @auvipy told "yeah gonna revert this PR. we need different approch for handling this" he also reviewed with comment "bigAutoField will be used for new django projects autogen primary key, that should not effect this package, for the change we had we should generate new migrations which will need to be supplied with this package.". Therefore, I concluded that the approach we are taking is to have have explicit ids for two models (rest already use explicit ids).

After writing this PR, it is reviewed that need to go back to the #495, therefore I suggest you "revert the revert of #495". Or if we want to go with this (explicit ids in models) approach we should work on this PR. And about the README correct, I will file new PR, if that is agreed upon.

It would not make sense to write an identical PR. Hope that makes sense.